src/test/java/com/worksap/nlp/sudachi/JapaneseTokenizerTest.java

```diff
@@ -256,6 +257,46 @@ public void disableEmptyMorpheme() throws IOException {
         assertThat(s.get(2).end(), is(1));
     }
 
+    @Test
+    public void splitAfterTokenizeCtoA() {
+        MorphemeList morphemesC = tokenizer.tokenize(Tokenizer.SplitMode.C, "東京都");
+        assertThat(morphemesC.size(), is(1));
+        MorphemeList morphemesA = morphemesC.splittedWith(Tokenizer.SplitMode.A);
+        assertThat(morphemesA.size(), is(2));
+    }
+
+    @Test
+    public void splitAfterTokenizeCtoB() {
+        MorphemeList morphemesC = tokenizer.tokenize(Tokenizer.SplitMode.C, "東京都");
+        assertThat(morphemesC.size(), is(1));
+        MorphemeList morphemesB = morphemesC.splittedWith(Tokenizer.SplitMode.B);
+        assertThat(morphemesB.size(), is(1));
+    }
+
+    @Test
+    public void splitAfterTokenizeCtoC() {
+        MorphemeList morphemes1 = tokenizer.tokenize(Tokenizer.SplitMode.C, "東京都");
+        assertThat(morphemes1.size(), is(1));
+        MorphemeList morphemes2 = morphemes1.splittedWith(Tokenizer.SplitMode.C);
+        assertThat(morphemes2, sameInstance(morphemes1));
+    }
+
+    @Test
+    public void splitAfterTokenizeAtoC() {
+        MorphemeList morphemes1 = tokenizer.tokenize(Tokenizer.SplitMode.A, "東京都");
+        assertThat(morphemes1.size(), is(2));
+        MorphemeList morphemes2 = morphemes1.splittedWith(Tokenizer.SplitMode.C);
+        assertThat(morphemes2, sameInstance(morphemes1));
+    }
+
+    @Test
+    public void splitAfterTokenizeBtoC() {
+        MorphemeList morphemes1 = tokenizer.tokenize(Tokenizer.SplitMode.B, "東京都");
+        assertThat(morphemes1.size(), is(1));
+        MorphemeList morphemes2 = morphemes1.splittedWith(Tokenizer.SplitMode.C);
+        assertThat(morphemes2, sameInstance(morphemes1));
+    }
+
```

Comment:
To be sure, please also run a test that produces a zero-length morpheme in the middle.

Reply:
Will do.